src/ripple/app/ledger/tests/common_ledger.cpp

```diff
@@ -527,6 +629,32 @@ verifyBalance(Ledger::pointer ledger, TestAccount const& account,
         "balance != amountReq");
 }
 
+void
+verifyLimit (Ledger::pointer ledger, TestAccount const& account,
+    Amount const& amount, std::uint32_t quality_in, std::uint32_t quality_out)
+{
+    auto sle = getLedgerEntryRippleState (ledger, account, amount.getIssuer (),
+        amount.getCurrency ());
+    if (sle == nullptr)
+        throw std::runtime_error ("!sle");
+
+    STAmount limit;
+    amountFromJsonNoThrow (limit, amount.getJson ());
+    if (limit != sle->getFieldAmount (sfHighLimit))
+        throw std::runtime_error("quality_in != HighQualityIn");
+
+    if (quality_in > 0 && quality_out > 0)
```

Comment:
Are `quality_in` and `quality_out` "all or nothing" fields? If not, then it would be better to check them separately, instead of in one block like this.